OpenRA.Mods.Common/Activities/PickupUnit.cs

```diff
@@ -69,25 +69,11 @@ public override bool Tick(Actor self)
 				return true;
 
 			if (IsCanceling)
-			{
-				if (carryall.State == Carryall.CarryallState.Reserved)
-					carryall.UnreserveCarryable(self);
-
-				// Make sure we run the TakeOff activity if we are / have landed
-				if (self.Trait<Aircraft>().HasInfluence())
-				{
-					ChildHasPriority = true;
-					IsInterruptible = false;
-					QueueChild(new TakeOff(self));
-					return false;
-				}
-
 				return true;
-			}
 
 			if (cargo.IsDead || carryable.IsTraitDisabled || !cargo.AppearsFriendlyTo(self))
 			{
-				carryall.UnreserveCarryable(self);
+				Cancel(self);
```

Comment:
Btw, this also means were losing orders queued by the user. (Not passing `keepQueue`.)